trieste/space.py

```diff
@@ -241,11 +247,48 @@ def __contains__(self, value: TensorType) -> bool | TensorType:
         return tf.reduce_all(value >= self._lower) and tf.reduce_all(value <= self._upper)
 
     def sample(self, num_samples: int) -> TensorType:
+        """
+        Sampling points randomly
+
+        :param num_samples: The number of points to sample from this search space.
+        :return: ``num_samples`` i.i.d. random points, sampled uniformly, and without replacement,
+            from this search space.
+        """
         dim = tf.shape(self._lower)[-1]
         return tf.random.uniform(
             (num_samples, dim), minval=self._lower, maxval=self._upper, dtype=self._lower.dtype
         )
 
+    def sample_halton(self, num_samples: int, seed: int = 0) -> TensorType:
+        """
+        Sampling points using randomized halton sequence
+
+        :param num_samples: The number of points to sample from this search space.
+        :param seed: Random seed for the halton sequence
+        :return: ``num_samples`` of points, using halton sequence.
+        """
+        if num_samples == 0:
+            return []
+        dim = tf.shape(self._lower)[-1]
+        return (self._upper - self._lower) * tfp.mcmc.sample_halton_sequence(
+            dim=dim, num_results=num_samples, dtype=self._lower.dtype, seed=seed
+        ) + self._lower
+
+    def sample_sobol(self, num_samples: int, skip: int = 0) -> TensorType:
+        """
+        Arange points using sobol sample
+
```

Comment:
Some comment similar to Halton. Can  you also comment why you would use one rather than the other?

Reply:
hmm, not sure. the difference between the two is, Halton here is stochastic, while Sobol that we used here is deterministic. we can also add details about two by referring to some paper like https://zancojournals.su.edu.krd/index.php/JPAS/article/view/1031 but not sure if it is needed here.